### PR TITLE
Breaking: Deprecate Readable

### PR DESCRIPTION
**Pull request by abdel-17**
- Removes the `Readable` utility in favor of `createSubscriber` from Svelte.
- Replaces the `useActiveElement` utility with a class-based `ActiveElement` utility.

Implementation notes:
- Extracted the logic for `testWithEffect` to a separate utility.
- The tests do `const current = $derived(activeElement.current)` to trigger a subscription. Otherwise, it's not considered in an effect, so the focus event handlers never get added.
- Removed `sleep` from tests where it's not needed.
- Removed the `focus` test utility. It's not needed for the tests to pass because DOM event handlers are executed synchronously. You only need to `flushSync` if you need to trigger pending effects.

### CODE REVIEW

**Comment by abdel-17** [comment]
The latest commit slightly changes how `ActiveElement` works with regards to Shadow DOM. It no longer recursively traverses the passing in `document` until it finds the active element. This simplifies internals and, unless I'm missing something, it makes more sense as the `document` prop specifies which document the utility should look inside.

```ts
import { activeElement, ActiveElement } from "runed";

const shadowActiveElement = new ActiveElement({
  document: shadowRoot,
});

const current = $derived(shadowActiveElement.current ?? activeElement.current);
```

**Comment by huntabyte** [comment]
> The latest commit slightly changes how ActiveElement works with regards to Shadow DOM. It no longer recursively traverses the passing in document until it finds the active element. This simplifies internals and, unless I'm missing something, it makes more sense as the document prop specifies which document the utility should look inside.

The reason we traverse is that, say, we don't pass a custom document right, meaning we want to get the `activeElement` of anything in the `document,` including things that may be inside of Shadow DOM. When that is the case, if the focus is within some `<button>` inside of the Shadow DOM, the `activeElement` will be that shadow root, rather than the `<button>` inside of it.

That's why if the `activeElement` is a shadow root, we then traverse it to find what the _true_ active element is.

---

The scenario when someone passes in a shadow root as the `document` would mean they _only_ want to track the active element _inside_ the shadow root.

File: sites/docs/src/content/utilities/active-element.md [hunk 1]
```diff
@@ -37,6 +33,21 @@ flexible API.
 </p>
 ```
 
+## Shadow DOM
```
Comment (huntabyte) [inline sites/docs/src/content/utilities/active-element.md:36]:
```suggestion
## Custom Document
```

File: sites/docs/src/content/utilities/active-element.md [hunk 2]
```diff
@@ -37,6 +33,21 @@ flexible API.
...
+
+If you need to track focus within a shadow root, you can pass the shadow root to the `ActiveElement`
```
Comment (huntabyte) [inline sites/docs/src/content/utilities/active-element.md:38]:
```suggestion
If you wish to scope the focus tracking with a custom document or shadow root, you can pass a `DocumentOrShadowRoot` to the `ActiveElement` options:
```

**Pull request merged**